openid-4-verifiable-presentations-1_0.md

```diff
@@ -1494,9 +1494,159 @@ issuers in Self-Sovereign Identity ecosystems using TRAIN</title>
+
+```json
+{
+  "client_id": "client.example.org",
+  ...
+  "response_type": "vp_token",
+  "nonce": "n-0S6_WzA2Mj",
+  "client_metadata": {...},
+  "presentation_definition": {...}
+}
+```
+
+Out of the Authorization Request parameters defined in [@!RFC6749] and (#vp_token_request), the following are supported with this profile: 
+
+* `client_id`
+* `client_id_scheme`
+* `response_type`
+* `nonce`
+* `presentation_definition`
+* `client_metadata`
+* `request`
+* `response_mode`
+
+The `client_id` and `client_id_scheme` MUST be omitted in unsigned requests defined in (#unsigned_request). The Wallet determines the Client Identifier from the origin as asserted by the Browser. How the Wallet receives the origin from is up to the user agent and app platform and is out of scope of this profile.
+
+The value of `response_mode` parameter MUST be `w3c_dc_api` when the response is not signed and/or encrypted and `w3c_dc_api.jwt` when the response is signed and/or encrypted as defined in (#jarm).
+
+In addition to the above-mentioned parameters, this profile introduces a following new parameter:
+
+* `expected_origins`: An array of strings, each of the string representing an origin of the Verifier making the request. This parameter MUST only be used with signed requests defined in (#signed_request). It relates the logical Client Identifier to the physical endpoints that are legit origins for requests on behalf of this Client Identifier and is used to detect request replay.
+
+## Signed and Unsigned Requests
+
+Any OpenID4VP request compliant to this specification can be sent over the W3C Digital Credentials API. Depending on the mechanism used to identify and authenticate the Verifier, the request can be signed or unsigned. This section defines signed and unsigned OpenID4VP requests over the W3C Digital Credentials API.
+
+### Unsigned Request {#unsigned_request}
+
+The Verifier MAY send all the OID4VP request data as JSON elements in the `request` API parameter and receives the result in the API's `result` parameter. In this case, the Wallet will use the Verifier origin as asserted by the Browser as the Verifer's Client Identifier.
+
+### Signed Request {#signed_request}
+
+The Verifier MAY send a signed request, for example, when identification and authentication of the Verifier is required.
+
+The signed Request Object MAY contain all the parameters listed in (#browser_api_request), except `request`. The signed Request Object MUST contain an `expected_origins` parameter. 
+
+Below is a non-normative example of such a request:
+
+```js
+const credential = await navigator.identity.get({
+  digital: {
+    providers: [{
+      protocol: "urn:openid.net:oid4vp",
+      request: {
+       client_id: "client.example.org",
+       client_id_scheme: "entity_id",
+       request: "eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJ..."
```

Comment:
This header has "typ":"JWT" which isn't really a good idea; suggested just truncating a bit more:

```suggestion
       request: "eyJhbGciOiJF..."
```

Reply:
fixed